custom/opm/utils.py

```diff
@@ -28,6 +29,7 @@ def _safeint(value):
         return 0
 
 
+@memoized
```

Comment:
The class level decorator works, it always returns same singleton. But this doesn't look easily understandable. I will change it to a function based one